src/index.ts

```diff
@@ -10,23 +10,60 @@ export { default as IgcCardContent } from './components/card/card.content';
 export { default as IgcCardHeader } from './components/card/card.header';
 export { default as IgcCardMedia } from './components/card/card.media';
 export { default as IgcCheckboxComponent } from './components/checkbox/checkbox';
-export { default as IgcFormComponent } from './components/form/form';
+export {
+  default as IgcFormComponent,
+  IgcFormEventMap,
+} from './components/form/form';
 export { default as IgcIconButtonComponent } from './components/button/icon-button';
 export { default as IgcIconComponent } from './components/icon/icon';
-export { default as IgcInputComponent } from './components/input/input';
+export {
+  default as IgcInputComponent,
+  IgcInputEventMap,
+} from './components/input/input';
 export { default as IgcLinkButtonComponent } from './components/button/link-button';
 export { default as IgcListComponent } from './components/list/list';
 export { default as IgcListHeaderComponent } from './components/list/list-header';
 export { default as IgcListItemComponent } from './components/list/list-item';
 export { default as IgcNavbarComponent } from './components/navbar/navbar';
-export { default as IgcNavDrawerComponent } from './components/nav-drawer/nav-drawer';
+export {
+  default as IgcNavDrawerComponent,
+  IgcNavDrawerEventMap,
+} from './components/nav-drawer/nav-drawer';
 export { default as IgcNavDrawerHeaderItemComponent } from './components/nav-drawer/nav-drawer-header-item';
 export { default as IgcNavDrawerItemComponent } from './components/nav-drawer/nav-drawer-item';
-export { default as IgcRadioComponent } from './components/radio/radio';
+export {
+  default as IgcRadioComponent,
+  IgcRadioEventMap,
+} from './components/radio/radio';
 export { default as IgcRadioGroupComponent } from './components/radio-group/radio-group';
 export { default as IgcRippleComponent } from './components/ripple/ripple';
 export { default as IgcSwitchComponent } from './components/checkbox/switch';
 
+export { SizableMixin } from './components/common/mixins/sizable';
+
+export {
+  IgcButtonBaseComponent,
+  IgcButtonEventMap,
+} from './components/button/button-base';
+
+export {
+  IgcCalendarBaseComponent,
+  IgcCalendarBaseEventMap,
+} from './components/calendar/common/calendar-base';
+export { IgcDaysViewEventMap } from './components/calendar/days-view/days-view';
+export {
+  DateRangeDescriptor,
+  DateRangeType,
+  TimeDeltaInterval,
+  WEEKDAYS,
+  Calendar,
```

Comment:
These are used only internally by the calendar.
```suggestion
  DateRangeDescriptor,
  DateRangeType,
```